packages/build/src/install/missing.js

```diff
@@ -21,16 +23,56 @@ export const installMissingPlugins = async function ({ missingPlugins, autoPlugi
   await addExactDependencies({ packageRoot: autoPluginsDir, isLocal: mode !== 'buildbot', packages })
 }
 
-export const installIntegrationPlugins = async function ({ integrations, autoPluginsDir, mode, logs }) {
-  const packages = integrations.map(getIntegrationPackage)
-  logInstallIntegrations(logs, integrations)
+export const installIntegrationPlugins = async function ({ integrations, autoPluginsDir, mode, logs, context }) {
+  const integrationsToBuild = integrations.filter(
+    (integration) => typeof integration.dev !== 'undefined' && context === 'dev',
+  )
+  if (integrationsToBuild.length) {
+    logSubHeader(logs, 'Building integrations')
+    logArray(
+      logs,
+      integrationsToBuild.map(({ slug, dev: { path } }) => `${slug} from ${path}`),
+    )
+  }
+  const packages = (
+    await Promise.all(integrations.map((integration) => getIntegrationPackage({ integration, context })))
+  ).filter(Boolean)
+  logInstallIntegrations(
+    logs,
+    integrations.filter((integration) =>
+      integrationsToBuild.every((compiledIntegration) => integration.slug !== compiledIntegration.slug),
+    ),
+  )
 
   await createAutoPluginsDir(logs, autoPluginsDir)
   await addExactDependencies({ packageRoot: autoPluginsDir, isLocal: mode !== 'buildbot', packages })
 }
 
-const getIntegrationPackage = function ({ version }) {
-  return `${version}/packages/buildhooks.tgz`
+const getIntegrationPackage = async function ({ integration: { version, dev }, context }) {
+  if (typeof version !== 'undefined') {
+    return `${version}/packages/buildhooks.tgz`
+  }
+
+  if (typeof dev !== 'undefined' && context === 'dev') {
+    const { path } = dev
+
+    const integrationDir = resolve(process.cwd(), path)
+    try {
+      const res = await execa('npm', ['run', 'build'], { cwd: integrationDir })
+
+      // This is horrible and hacky, but `npm run build` will
+      // return status code 0 even if it fails
+      if (!res.stdout.includes('Build complete!')) {
```

Comment:
Where is this `npm run build` coming from? Is it something we control, and if so, could we fix the status code?

Also, where is this "Build complete!" string being generated? Referencing it here might help (very slightly) reduce the chances of someone changing this string unaware that it will break the detection mechanism here.

Reply:
Agree on this. We have logged an issue so that we can fast follow this post launch: https://github.com/netlify/sdk/issues/635

Reply:
For context, that `npm run build` is executing the `build` script of an integration, which calls the SDK's build tool. Fully in our domain, just haven't got the capacity to investigate immediately